librz/analysis/arch/arm/arm_il32.c

```diff
@@ -2314,22 +2515,880 @@ static RzILOpEffect *tbb(cs_insn *insn, bool is_thumb) {
+	} else {
+		wback_eff = NOP();
+	}
+
+	return SEQ2(eff, wback_eff);
+}
+
+static RzILOpEffect *vldn_all_lane(cs_insn *insn, bool is_thumb) {
+	ut32 rm_idx = OPCOUNT() - 1;
+	ut32 rn_idx;
+	bool use_rm = false;
+	ut32 regs; // number of regs in {list}
+
+	if (!ISMEM(rm_idx)) {
+		use_rm = true;
+		regs = OPCOUNT() - 2;
+	} else {
+		rm_idx = -1;
+		regs = OPCOUNT() - 1;
+	}
+	rn_idx = regs;
+
+	ut32 n = insn->id - ARM_INS_VLD1 + 1;
+	if (n != regs) {
+		return NULL;
+	}
+
+	RzILOpBitVector *data0 = NULL, *data1 = NULL, *data2 = NULL, *data3 = NULL;
+	RzILOpEffect *eff = NULL;
+	RzILOpBitVector *addr = ARG(rn_idx);
+	ut32 elem_bits = VVEC_SIZE(insn);
+	ut32 elem_bytes = elem_bits / 8;
+	ut32 dreg_size = 64;
+	switch (n) {
+	case 1:
+		data0 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		eff = write_reg(REGID(0), DUP(data0));
+		if (regs == 2) {
+			eff = SEQ2(eff, write_reg(REGID(1), data0));
+		}
+		break;
+	case 2:
+		data0 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data1 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		eff = SEQ2(write_reg(REGID(0), data0),
+			write_reg(REGID(1), data1));
+		break;
+	case 3:
+		data0 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data1 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data2 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		eff = SEQ3(write_reg(REGID(0), data0),
+			write_reg(REGID(1), data1),
+			write_reg(REGID(2), data2));
+		break;
+	case 4:
+		data0 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data1 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data2 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		addr = ADD(DUP(addr), UN(8, elem_bytes));
+		data3 = replicated_val(elem_bits, dreg_size, LOADW(elem_bits, addr));
+		eff = SEQ4(write_reg(REGID(0), data0),
+			write_reg(REGID(1), data1),
+			write_reg(REGID(2), data2),
+			write_reg(REGID(3), data3));
+		break;
+	default:
+		rz_warn_if_reached();
+		return NULL;
+	}
+
+	bool wback = insn->detail->arm.writeback;
+	RzILOpEffect *wback_eff;
+	if (wback) {
+		RzILOpBitVector *new_offset = use_rm ? ARG(rm_idx) : UN(32, elem_bytes * n);
+		wback_eff = write_reg(rn_idx, ADD(REG_VAL(rn_idx), new_offset));
+	} else {
+		wback_eff = NOP();
+	}
+
+	return SEQ2(eff, wback_eff);
+}
+
+static RzILOpEffect *vldn(cs_insn *insn, bool is_thumb) {
+	if (OPCOUNT() < 2 || !ISREG(0)) {
+		return NULL;
+	}
+
+	// to single lane
+	if (NEON_LANE(0) != -1) {
+		return vldn_single_lane(insn, is_thumb);
+	}
+
+	// capstone cannot distinguish details of the following instructions
+	// vld3.8 {d0, d1, d2}, [r0] (f420040f)
```

Comment:
This one should be addressed by https://github.com/capstone-engine/capstone/pull/1949
Please add `TODO` or `FIXME` to update once we merge our `capstone-ng` branch. cc @Rot127

Reply:
Unfortunately not. Capstone assigns instruction IDs by their Mnemonic (which is nice, because we do not have to deal with the original >4000 IDs). But this means that both of them are classified as `ARM_INS_VLD3`.

The LLVM id is not accessible from the CS API. We need to extend the API for such cases. It is related to https://github.com/capstone-engine/capstone/issues/1945. Access to the raw TableGen info is requested there as well.